board/miyoo/boot/firstboot

```diff
@@ -548,21 +540,14 @@ if (dialog --clear --stdout --ok-label YES --cancel-label NO --title "   AUTO-RE
 	echo "The full size of your SD is ${SD_SIZE_IN_MiB}MB large" | tee -a ${LOG}
 	echo "Unallocated space available for resizing equals ${SD_MAX_UNALL_SIZE_IN_GiB}GB," | tee -a ${LOG}
 	echo "and more precisely it's ${SD_MAX_UNALL_SIZE_IN_MiB}MB big" | tee -a ${LOG}
-	if ! $RESIZE_ABORT; then
-		echo "Resizing the MAIN partition to ${SD_MAX_AVAIL_SIZE_IN_GiB}GB." | tee -a ${LOG}
-		echo "This can take A LONG TIME, so be patient." | tee -a ${LOG}
-		umount /dev/mmcblk0p4 | tee -a ${LOG}
-		fsck.fat -y /dev/mmcblk0p4 | tee -a ${LOG}
-		#TODO: fix outputting logs from fatresize stdout
-		fatresize --size="${SD_MAX_AVAIL_SIZE_IN_KiB}"ki -v -p /dev/mmcblk0p4 -n4 | tee -a ${LOG}
-		fsck.fat -y /dev/mmcblk0p4 | tee -a ${LOG}
-		echo -n -e \\xEB\\x58\\x90 > /dev/mmcblk0p4
-		sync
-		mount -t vfat -o rw,sync,utf8 /dev/mmcblk0p4 ${HOME} | tee -a ${LOG}
-	else
-		echo "MAIN fat partition resizing was aborted!" | tee -a ${LOG}
-		sleep 2
-	fi
+	echo "Resizing the MAIN partition to ${SD_MAX_AVAIL_SIZE_IN_GiB}GB." | tee -a ${LOG}
+	echo "This can take a few seconds." | tee -a ${LOG}
+	umount /dev/mmcblk0p4 | tee -a ${LOG}
+	parted -a optimal /dev/mmcblk0 --script rm 4  | tee -a ${LOG}
+	parted -a optimal /dev/mmcblk0 --script mkpart primary 669MB 100%  | tee -a ${LOG}
```

Comment:
Use non arbitrary value for start point, smth like:
`START_POINT=(${P1_SIZE} + ${P2_SIZE} + ${P3_SIZE} + ${P4_SIZE}) + $BOOT_SIZE` but in MB